timeseries/src/autogluon/timeseries/models/gluonts/callback.py

```diff
@@ -37,3 +37,26 @@ def on_epoch_end(
                 logger.warning("Time limit exceed during training, stop training.")
                 return False
         return True
+
+
+class EarlyStoppingCallback(Callback):
+    """GluonTS callback to early stop the training if the validation loss
+    is not improved for `patience' round. For the GluonTS models used in autogluon,
+    the loss is always minimized."""
+
+    def __init__(self, patience=10):
```

Comment:
We need to be careful here, this is not a good patience value to use in all situations. This is a complex topic and I put a lot of effort into this in Tabular.

For example, I have Early stopping logic coded here: https://github.com/awslabs/autogluon/blob/master/core/src/autogluon/core/utils/early_stopping.py

In particular, AdaptiveES will be much better than naive patience based ES. But regardless, we should err on the side of caution and use high patience values rather than low ones, otherwise we risk crippling the models in situations where time isn't much of a concern.

Reply:
Thanks for the comments!

I am not very sure about 10 is not good for *all* situations but I agree that we need some data points to set this default value. Do you have some results to share about the adaptive early stopping?

I also implemented the adaptive early stopping logic for GluonTS model, there was quite some code duplicates and I am thinking of talking the `_update_patience` out of the class `AdaptiveES` so I could reuse it. Maybe you have better suggestion.

I also remove the default usage of early stopping for now. After we have more results for it, we could set the default option later based on that. Let me know what do you think of this option.

Reply:
Sorry for confusion, when I said "all situations" I meant that in some situations it might not be good, not that it is always not good.

Regarding your implementation, to avoid code dupe why not directly use the AdaptiveES code? You can import it and use it in your callback to avoid code dupe. I do this for individual models, for example, having it as `self.es`: https://github.com/awslabs/autogluon/blob/master/tabular/src/autogluon/tabular/models/xgboost/callbacks.py#L27-L30

We probably shouldn't remove default usage of early stopping, just err on the side of caution and use a larger value. What value is best will be understood after we start doing benchmarking. Maybe 10 is fine but it depends on the model.

Reply:
perhaps even 5 is fine if we have it on by default. if anything it's a little too cautious considering that we work with 50-100 epochs.

Reply:
Re the code duplicates, I followed your suggestions. For the default patience, I put it to 20 for now. We may change it later with more data points.